client/ayon_hiero/api/lib.py

```diff
@@ -622,9 +632,7 @@ def get_publish_attribute(tag):
     """
     tag_data = tag.metadata()
     # get data to the publish attribute
-    value = tag_data.value("tag.publish")
-    # return value converted to bool value. Atring is stored in tag.
-    return ast.literal_eval(value)
+    return json.loads(tag_data.get("tag.json_metadata"))["publish"]
```

Comment:
Same as above. Error on key error instead of on `json.loads` if key does not exist.

```suggestion
    return json.loads(tag_data["tag.json_metadata"])["publish"]
```
That is assuming this actually works on whatever `tag.metadata()` returns as object.

Reply:
I think this code was indeed a bit convoluted and initially caused by `tag.metadata()` returning a `hiero.core.Metadata` object. But you're right I've simplified it in something that hopefully makes more sense.